src/basic_memory/mcp/tools/build_context.py

```diff
@@ -35,7 +35,7 @@
 )
 async def build_context(
     url: MemoryUrl,
-    depth: Optional[int] = 1,
+    depth: Optional[Union[int, str]] = 1,
```

Comment:
In my experience, adding more types to the mcp args will sometimes mess up tools like vscode, but lets give this a shot. It's more "correct". FYI, I prefer `Optional[int | str]` or even a type alias
```python
type StringOrInt = int | str
depth: Optional[StringOrInt]
```

Reply:
Cool